### PR TITLE
Feature/as 117 date error improvements

### PR DESCRIPTION
**Pull request by davehlewis**
## Ticket Number

AS-117

## Description of change
Updated the validators as per the ticket
- Updated feature file and steps accordingly
- Created new middleware to combine the date inputs into `req.body`
- Created reusable generic validator for date inputs that checks for valid date and highlights errors as required by GDS

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [X] My commit history in this PR is linear
- [X] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

**Review by rcmsoftwaredevelopment — APPROVED** [review]

**Review by rcmsoftwaredevelopment — DISMISSED** [review]

**Review by danWatkinson — DISMISSED** [review]
cool. agree that the table structuring is as we spoke about in 3 amigos
couple of notes (not challenges ;) )
* agree it would be better to do a 'component' that understood the mechanical date stuff, compromise accepted
* kinda wrapped up in that- we like our BDD stuff to be more abstract.. again- if we popped the mechanical date stuff out i think that would be easier... again- compromise accepted.

File: packages/forms-web-app/tests/unit/validators/custom/date-input.test.js [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+      const rule = dateInputValidation(mockId, mockLabel)[4].builder.build();
+
+      expect(rule.fields).toEqual(['mock-date-month']);
+      expect(rule.locations).toEqual(['body']);
+      expect(rule.optional).toBeFalsy();
+      expect(rule.stack[0].options).toEqual([{ min: 1, max: 12 }]);
+      expect(rule.stack[0].negated).toBeFalsy();
+    });
+
+    it(`has a rule for empty day input`, () => {
```
Comment (a6software) [inline packages/forms-web-app/tests/unit/validators/custom/date-input.test.js:61]: test name is confusing - same for lines 51, and 40

Comment (a6software) [reply packages/forms-web-app/tests/unit/validators/custom/date-input.test.js:61]: rest looks good to me 👍

Comment (davehlewis) [reply packages/forms-web-app/tests/unit/validators/custom/date-input.test.js:61]: Unfortunately there are 2 sets of rules for each input so this seemed the best way to split them. First deals with if they are empty, second deals with if they are valid

**Comment by davehlewis** [comment]
Needs another re-review due to changes to how commands are written

**Review by rcmsoftwaredevelopment — APPROVED** [review]

**Review by Fenisay — APPROVED** [review]
LGTM

**Pull request merged**